Update backup instructions (#531)
cert-manager/website: content/en/docs/tutorials/backup.md

$ kubectl apply -f cert-manager-backup.yaml
```

## Full cluster backup and restore

This section refers to backing up and restoring 'all' Kubernetes resources in a
cluster — including some `cert-manager` ones — for scenarios such as disaster
recovery, cluster migration etc.

*Note*: We have tested this process on simple Kubernetes test clusters with a limited set of Kubernetes releases. To avoid data loss, please test both the backup and the restore strategy on your own cluster before depending upon it in production. If you encounter any errors, please open a GitHub issue or a PR to document variations on this process for different Kubernetes environments. 

### Avoiding unnecessary certificate reissuance

#### Order of restore

If `cert-manager` does not find a Kubernetes `Secret` with an X.509 certificate
for a `Certificate`, reissuance will be triggered. To avoid unnecessary
reissuance after a restore, ensure that `Secret`s are restored before
`Certificate`s. Similarly, `Secret`s should be restored before `Ingress`es if you
are using [`ingress-shim`](../../usage/ingress/).

#### Excluding some cert-manager resources from backup

> `cert-manager` has a number of resources- `Order`s, `Challenge`s and
> `CertificateRequest`s -that are designed to represent a point-in-time operation
> such as a request for a certificate. As such, their status often depends on other
> ephemeral resources (i.e a temporary Secret holding a private key), so
> `cert-manager` cannot always correctly re-create the status of these resources
> by just looking at the cluster state at some later time.

Reviewer: ```suggestion
`cert-manager` has a number of resources — `Order`s, `Challenge`s and
`CertificateRequest`s  — that are designed to represent a point-in-time operation
such as a request for a certificate. As such, their status often depends on other
ephemeral resources (i.e a temporary Secret holding a private key), so
`cert-manager` cannot always correctly re-create the status of these resources
by just looking at the cluster state at some later time.
```

Author: Thanks @wallrj ! I have eliminated those dashes by re-writing this using shorter separate sentences here and in a few more places 😄

$ kubectl apply -f cert-manager-backup.yaml
```

## Full cluster backup and restore

This section refers to backing up and restoring 'all' Kubernetes resources in a
cluster- including some `cert-manager` ones- for scenarios such as disaster
recovery, cluster migration etc.

> Note that we have not done any extensive testing, only a few common scenarios-
> it is recommended to test the backup and restore strategy before putting it in place
> as there are likely to be some other edge cases.
> We would like to ensure that users can reliably backup and restore `cert-manager`
> resources using common tools- if you encounter any errors, please do open a
> GitHub issue or PR an update to this document.

Reviewer: ```suggestion
*Note*: We have tested this process on simple Kubernetes test clusters with a limited set of Kubernetes releases. To avoid data loss, please test both the backup and the restore strategy on your own cluster before depending upon it in production. If you encounter any errors, please open a GitHub issue or a PR to document variations on this process for different Kubernetes environments. 
```

I suggest something like that ^^  instead, but happy to go with the original version if you prefer.

### Avoiding unnecessary certificate reissuance

#### Order of restore

> If `cert-manager` does not find a Kubernetes Secret with an X.509 certificate
> for a `Certificate`, reissuance will be triggered. To avoid unnecessary
> reissuance after a restore, ensure that Secrets are restored before
> `Certificate`s. Similarly, Secrets should be restored before `Ingress`es if you
> are using [`ingress-shim`](../../usage/ingress/).

Reviewer: ```suggestion
If `cert-manager` does not find a Kubernetes `Secret` with an X.509 certificate
for a `Certificate`, reissuance will be triggered. To avoid unnecessary
reissuance after a restore, ensure that `Secret`s are restored before
`Certificate`s. Similarly, `Secret`s should be restored before `Ingress`es if you
are using [`ingress-shim`](../../usage/ingress/).
```

Maybe quote `Secret` for consistency with the quoting of other Kubernetes resource types.

#### Excluding some cert-manager resources from backup

`cert-manager` has a number of resources- `Order`s, `Challenge`s and
`CertificateRequest`s -that are designed to represent a point-in-time operation
such as a request for a certificate. As such, their status often depends on other
ephemeral resources (i.e a temporary Secret holding a private key), so
`cert-manager` cannot always correctly re-create the status of these resources
by just looking at the cluster state at some later time.

In most cases backup and restore tools will not restore the statuses of custom resources,
so including such one-time resources in a backup can result in an unnecessary reissuance
after a restore as without the status fields `cert-manager` will not be able to tell that,
for example, an `Order` has already been fulfilled.
To avoid unnecessary reissuance, we recommend that `Order`s and `Challenge`s are excluded

> from the backup. We also don't recommend backing up `CertificateRequest`s, see [Backing up CertificateRequests](#backing-up-certificaterequests)

Reviewer: That's a nice explanation.
